pkg/models/terminal/terminal.go

```diff
@@ -140,15 +151,135 @@ func (t TerminalSession) Close(status uint32, reason string) {
 
 type Interface interface {
 	HandleSession(shell, namespace, podName, containerName string, conn *websocket.Conn)
+	HandleShellAccessToNode(nodename string, conn *websocket.Conn)
 }
 
 type terminaler struct {
-	client kubernetes.Interface
-	config *rest.Config
+	client  kubernetes.Interface
+	config  *rest.Config
+	options *Options
+}
+
+type NodeTerminaler struct {
+	Nodename      string
+	Namespace     string
+	PodName       string
+	ContainerName string
+	Shell         string
+	Privileged    bool
+	Config        *Options
+	client        kubernetes.Interface
+}
+
+func NewTerminaler(client kubernetes.Interface, config *rest.Config, options *Options) Interface {
+	return &terminaler{client: client, config: config, options: options}
+}
+
+func NewNodeTerminaler(nodename string, options *Options, client kubernetes.Interface) (*NodeTerminaler, error) {
+
+	n := &NodeTerminaler{
+		Namespace:     "kubesphere-controls-system",
+		ContainerName: "nsenter",
+		Nodename:      nodename,
+		PodName:       nodename + "-shell-access",
+		Shell:         "sh",
+		Privileged:    true,
+		Config:        options,
+		client:        client,
+	}
+
+	node, err := n.client.CoreV1().Nodes().Get(context.Background(), n.Nodename, metav1.GetOptions{})
+
+	if err != nil {
+		return n, fmt.Errorf("node cannot exist. nodename:%s, err: %v", n.Nodename, err)
+	}
+
+	flag := false
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady && condition.Status == v1.ConditionTrue {
+			flag = true
```

Comment:
Need a break here? Even if it doesn't seem to affect anything.

Reply:
thanks, i'll modify it

Reply:
I think we need to filter out unschedulable nodes, if user cordon nodes, it's unlikely to create pod on these nodes.